### ISSUE TITLE
Camera doesn't focus

### ISSUE THREAD

**Issue by jans23** [M1cha/android_device_xiaomi_aries#56]
The camera can't focus at all and often shows the green indicator as if it has focused when really the image is terribly blurry. Also close up pics are impossible. This is true for the stock camera app and Xiaomi's camera app. Any help would be much appreciated.

**Comment by dnle71** [comment]
Yes! compared to mokee 4.4, this is awful

**Comment by dnle71** [comment]
Also, please do look at the mi2/s specific mokee commits for fixes. Pretty please.

**Comment by M1cha** [comment]
which commits?

**Comment by M1cha** [comment]
also pls tell me which camera lense you have because I can't reproduce the problem

**Comment by jans23** [comment]
I have an Mi2S. How can I find out details about my camera lense?

Samsung Galaxy S3 users seem to have a similar problem. Maybe their fix is useful here as well. See: http://forum.xda-developers.com/galaxy-s3/themes-apps/17-fixed-camera-app-lib-cm-aosp-roms-t2505973

**Comment by dnle71** [comment]
I don't know how to code but, Im sure you could some useful fixes here. 

https://github.com/MoKee/android_device_xiaomi_aries

And here's my hwinfo.

http://xiaomi.eu/community/index.php?posts/212725

**Comment by dnle71** [comment]
Im sorry, I tried mokee earlier and ever since it got to 4.4.4 the camera got blurry and the phone number is unknown too. :(

**Comment by jans23** [comment]
Any update on this issue? The camera is absolutely unusable which is why this issue prevents me from switching to CM permanently. Please fix it.

**Comment by M1cha** [comment]
You have to provide logs.

**Comment by jans23** [comment]
I will. Could you point me to instructions which logs you need and how 
to collect them? Thanks.

**Comment by M1cha** [comment]
I need logcat and dmesg.
You can use this app: https://play.google.com/store/apps/details?id=org.l6n.sendlog
Also pls note that u have to dump them right after the failure.(minutes not seconds ;) )

**Comment by jans23** [comment]
Here are the logs. Please let me know if you need anything else.
http://pastebin.com/E5FSthtW
http://pastebin.com/sEnwJtmu

**Comment by DragooNArT** [comment]
Hi M1cha,

I can also confirm that there is some kind of issue with regards to camera focus,
I'm using Mi2s 32GB with the 13mp camera and the latest cm11 build.

Symptoms: Camera pre-focuses correctly and is in focus during preview, however
the picture that is taken does not have the same focus point as the preview does.

In simpler terms the in-focus preview ends up as a slightly out of focus picture.
This behavior occurs on all camera apps(not just the default one).

It would be great if we can get some feedback on what the progress is with this issue :)

Regards,
Nikolay

**Comment by goranius** [comment]
I have the same issue with Mi2S 32GB with the 13 mp camera cm11 build of 20141030-NIGHTLY kernel 3.4.0-cyanogenmod-ge27b401 build02@cyanogenmod #1, if you need more information tell me how can I get t and I send to you.
Regards
